OSDOCS-11141 OCP 4.16.1 Release Notes (#78379)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

$ oc adm release info 4.16.0 --pullspecs
----
//replace 4.y.z for the correct values for the release. You do not need to update oc to run this command.

[id="ocp-4-16-1_{context}"]
=== RHSA-2024:4156 - {product-title} {product-version}.1 bug fix and security update

Issued: 2024-07-02

{product-title} release {product-version}.1, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:4156[RHSA-2024:4156] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:4159[RHSA-2024:4159] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.16.1 --pullspecs
----

[id="ocp-4-16-1-bug-fixes_{context}"]
==== Bug fixes

> * Previously, an error in `growpart` caused the device to be locked, which prevented the Linux Unified Key Setup-on-disk-format (LUKS) device being opened. The resulted was that the system was unable to boot and went into emergency mode. With this release, the call to the `growpart` is removed and this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-35973[*OCPBUGS-35973*])

Reviewer: ```suggestion
* Previously, an error in `growpart` caused the device to be locked, which prevented the Linux Unified Key Setup-on-disk-format (LUKS) device from being opened. As a result, the system was unable to boot and went into emergency mode. With this release, the call to the `growpart` is removed and this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-35973[*OCPBUGS-35973*])
```

Reviewer: Also, s/system/node ?
```suggestion
* Previously, an error in `growpart` caused the device to be locked, which prevented the Linux Unified Key Setup-on-disk-format (LUKS) device from being opened. As a result, the node was unable to boot and went into emergency mode. With this release, the call to the `growpart` is removed and this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-35973[*OCPBUGS-35973*])
```

> * Previously, a bug in systemd might cause the `coreos-multipath-trigger.service` unit to hang forever. As a result the system would never finish booting. With this release, the systemd unit was removed and the issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-35748[*OCPBUGS-35748*])

Reviewer: ```suggestion
* Previously, a bug in systemd might have caused the `coreos-multipath-trigger.service` unit to hang indefinitely. As a result, the system would never finish booting. With this release, the systemd unit was removed and the issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-35748[*OCPBUGS-35748*])
```

> * Previously, the KMS key was applied as an empty string. This always applied a KMS key that was invalid. With this release, the empty string is removed and the KMS key is only applied when one exists from the `install-config.yaml`. (link:https://issues.redhat.com/browse/OCPBUGS-35531[*OCPBUGS-35531*])

Reviewer: ```suggestion
* Previously, the KMS key was applied as an empty string, which caused the key to be invalid. With this release, the empty string is removed and the KMS key is only applied when one exists from the `install-config.yaml`. (link:https://issues.redhat.com/browse/OCPBUGS-35531[*OCPBUGS-35531*])
```

* Previously, there was no validation of the values for confidential compute and on host maintenance set by the user. With this release, when confidential compute is enabled by the user the value for `onHostMaintenance` must be set to `onHostMaintenance: Terminate`. (link:https://issues.redhat.com/browse/OCPBUGS-35493[*OCPBUGS-35493*])

> * Previously, in User Provisioned Infrastructure (UPI) or clusters that were upgraded from older versions, `failureDomains` may be missing in Infrastructure objects which caused certain checks to fail. With this release, a fallback `failureDomains` is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35446[*OCPBUGS-35446*])

Reviewer: ```suggestion
* Previously, in user-provisioned infrastructure  (UPI) clusters or clusters that were upgraded from older versions, `failureDomains` might be missing in Infrastructure objects, which caused certain checks to fail. With this release, a `failureDomains` fallback is synthesized from `cloudConfig` if none are available in `infrastructures.config.openshift.io`. (link:https://issues.redhat.com/browse/OCPBUGS-35446[*OCPBUGS-35446*])
```

> * Previously, when a new version of a Custom Resource Definition (CRD) specified a new conversion strategy, this conversion strategy was expected to successfully convert resources. This was not the case because Operator Lifecycle Manager (OLM) cannot run the new conversion strategies for CRD validation without actually performing the update operation. With this release, the OLM implementation generates a warning message during the update process when CRD validations fail with the existing conversion strategy and the new conversion strategy is specified in the new version of the CRD. (link:https://issues.redhat.com/browse/OCPBUGS-35373[*OCPBUGS-35373*])

Reviewer: ```suggestion
* Previously, when a new version of a custom resource definition (CRD) specified a new conversion strategy, this conversion strategy was expected to successfully convert resources. This was not the case because Operator Lifecycle Manager (OLM) cannot run the new conversion strategies for CRD validation without actually performing the update operation. With this release, the OLM generates a warning message during the update process when CRD validations fail with the existing conversion strategy and the new conversion strategy is specified in the new version of the CRD. (link:https://issues.redhat.com/browse/OCPBUGS-35373[*OCPBUGS-35373*])
```

* Previously, Amazon Web Services (AWS) HyperShift clusters leveraged their Amazon Virtual Private Cloud (VPC)'s primary classless inter-domain routing (CIDR) range to generate security group rules on the data plane. As a consequence, installing AWS HyperShift clusters into an AWS VPC with multiple CIDR ranges could cause the generated security group rules to be insufficient. With this update, security group rules are generated based on the provided Machine CIDR range to resolve this issue. (link:https://issues.redhat.com/browse/OCPBUGS-35056[*OCPBUGS-35056*])

> * Previously, a change on what the Azure API returns for a subnet would cause the Installer to crash. With this release, the code can handle both the old and new data for subnets as well as returning an error in case the expected information is not found. (link:https://issues.redhat.com/browse/OCPBUGS-35023[*OCPBUGS-35023*])

Reviewer: ```suggestion
* Previously, a change to the field that Azure API returns for a subnet would cause the Installer to crash. With this release, the code can handle both the old and new data for subnets as well as returning an error in case the expected information is not found. (link:https://issues.redhat.com/browse/OCPBUGS-35023[*OCPBUGS-35023*])
```

> * Previously, the Source-to-Image (S2I) build strategy needed to be explicitly mentioned in the `func.yaml` in order to create Serverless function. Additionally, the error message did not indicate the problem. With this release, if S2I is not mentioned users can still create the function. However, if it is not S2I, users cannot create the function. Additionally, the error messages have been updated to provide more information. (link:https://issues.redhat.com/browse/OCPBUGS-34717[*OCPBUGS-34717*])

Reviewer: ```suggestion
* Previously, the Source-to-Image (S2I) build strategy needed to be explicitly added to the `func.yaml` in order to create the Serverless function. Additionally, the error message did not indicate the problem. With this release, if S2I is not added, users can still create the Serverless function. However, if it is not S2I, users cannot create the function. Additionally, the error messages have been updated to provide more information. (link:https://issues.redhat.com/browse/OCPBUGS-34717[*OCPBUGS-34717*])
```

> * Previously, the `CurrentImagePullSecret` field on the `MachineOSConfig` was not being used in the rollout process. With this release, the `CurrentImagePullSecret` field on the `MachineOSConfig` object is allowed to be used by the image rollout process. (link:https://issues.redhat.com/browse/OCPBUGS-34261[*OCPBUGS-34261*])

Reviewer: ```suggestion
* Previously, the `CurrentImagePullSecret` field on the `MachineOSConfig` object was not being used in when rolling out new on-cluster layering build images.. With this release, the `CurrentImagePullSecret` field on the `MachineOSConfig` object is allowed to be used by the image rollout process. (link:https://issues.redhat.com/browse/OCPBUGS-34261[*OCPBUGS-34261*])
```
